src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using System.Management.Automation;
+using System.Security.Cryptography;
+using System.Collections.Generic;
+using System.Reflection;
+using System.IO;
+
+namespace Microsoft.PowerShell
+{
+    /// <summary>
+    /// send up telemetry for startup
+    /// </summary>
+    public class ApplicationInsightsTelemetry
+    {
+        // The semaphore file which indicates whether telemetry should be sent
+        // This is temporary code waiting on the acceptance and implementation of the configuration spec
+        /// <summary>
+        /// The name of the file by when present in $PSHOME will enable telemetry.
+        /// If this file is not present, no telemetry will be sent.
+        /// </summary>
+        public const string TelemetrySemaphoreFilename = "DELETE_ME_TO_DISABLE_CONSOLEHOST_TELEMETRY";
+
+        /// <summary>
+        /// The path to the semaphore file which enables telemetry
+        /// </summary>
+        public static string TelemetrySemaphoreFilePath = Path.Combine(
+            Utils.GetApplicationBase(Utils.DefaultPowerShellShellID),
+            TelemetrySemaphoreFilename);
+
+        // Telemetry client to be reused when we start sending more telemetry
+        private static TelemetryClient _telemetryClient = null;
+
+        // Set this to true to reduce the latency of sending the telemetry
+        private static bool _developerMode = false;
+
+        // PSCoreInsight2 telemetry key
+        private const string _psCoreTelemetryKey = "ee4b2115-d347-47b0-adb6-b19c2c763808";
+
+        /// <summary>
+        /// Send the telemetry
+        /// </summary>
+        private static void SendTelemetry(string eventName, Dictionary<string,string>payload)
+        {
+            try
+            {
+                // if the semaphore file exists, try to send telemetry
+                if (Utils.NativeFileExists(TelemetrySemaphoreFilePath))
+                {
+                    TelemetryConfiguration.Active.InstrumentationKey = _psCoreTelemetryKey;
+                    TelemetryConfiguration.Active.TelemetryChannel.DeveloperMode = _developerMode;
```

Comment:
right now, it's a one-time thing. I'll create a static constructor and move this inside

---
In reply to: [113083840](https://github.com/PowerShell/PowerShell/pull/3620#discussion_r113083840) [](ancestors = 113083840)